adresse/controllers/service.classic.php

```diff
@@ -165,4 +169,95 @@ function update(){
     $rep->data = array('success' => ''.$result, 'type'=>$typeRes, 'message' => $message);
     return $rep;
   }
+
+  function export(){
+
+    $test='#^[0-9]{1}[0-9AB]{1}[0-9]{3}$#';
+    $filterParams = array();
+    $project = $this->param('project');
+    $repository = $this->param('repository');
+    $insee = $this->param('insee');
+    $option = $this->param('opt');
+
+    if(!$project){
+      $rep = $this->getResponse('json');
+      $rep->data = array('status'=>'error', 'message'=>'Project not find');
+      return $rep;
+    }
+
+    if(!$repository){
+      $rep = $this->getResponse('json');
+      $rep->data = array('status'=>'error', 'message'=>'Repository not find');
+      return $rep;
+    }
+
+    if(!$insee){
+      $rep = $this->getResponse('json');
+      $rep->data = array('status'=>'error', 'message'=>'Code insee not find');
+      return $rep;
+    }
+    if(!preg_match($test, $insee)){
+      $rep = $this->getResponse('json');
+      $rep->data = array('status'=>'error', 'message'=>'Code insee not avaible');
+      return;
+    }
+    $filterParams[] = $insee;
+
+    if(!$option){
+      $rep = $this->getResponse('json');
+      $rep->data = array('status'=>'error', 'message'=>'Option not find');
+      return $rep;
+    }
+
+    // vérifier que le repository et le project correspondent à un projet lizmap
+
+    $p = lizmap::getProject($repository.'~'.$project);
+    if( !$p ){
+      $rep = $this->getResponse('json');
+        $rep->data = array('status'=>'error', 'message'=>'A problem occured while loading project with Lizmap');
+        return $rep;
+    }
+
+    if (!$p->checkAcl()) {
+      $rep = $this->getResponse('json');
+        $rep->data = array('status'=>'error', 'message'=>jLocale::get('view~default.repository.access.denied'));
+        return $rep;
+    }
+
+    // vérifier que le projet contient la couche point_adresse
+
+    // demander la voie éditable à proximité de la geom
+
+    $autocomplete = jClasses::getService('adresse~search');
+    try {
+        $result = $autocomplete->getData( $repository, $project, 'point_adresse', $filterParams, $option);
+    } catch (Exception $e) {
+        $result = Null;
+    }
+    $leBal = jClasses::getService('adresse~exportBal');
+    $tempPath = jApp::tempPath('export');
+
+    jFile::createDir($tempPath);
+
+    $fileName = tempnam($tempPath, 'exportbal-');
+
+    if($result != Null){
+      $leBal->exportCSV($fileName, $result);
+
+    }else {
+      $rep = $this->getResponse('json');
+      $rep->data = array('status'=>'error', 'message'=>'Aucun résultat trouvé');
+      return $rep;
+    }
+    $resp = $this->getResponse('binary');
+
+    $resp->deleteFileAfterSending = true;
+    $resp->fileName = $fileName;
+    $resp->outputFileName = 'nom-sympa-pour-lutilisateur.csv';
+    $resp->mimeType = 'text/csv';
+    $resp->doDownload = true; // true si tu veux que l'utilisateur ait une boite de dialogue "sauver sous"
+
+    return $resp;
```

Comment:
```suggestion
    $rep = $this->getResponse('binary');

    $rep->deleteFileAfterSending = true;
    $rep->fileName = $fileName;
    $rep->outputFileName = 'nom-sympa-pour-lutilisateur.csv';
    $rep->mimeType = 'text/csv';
    $rep->doDownload = true; // true si tu veux que l'utilisateur ait une boite de dialogue "sauver sous"

    return $rep;
```
Pour être homogène avec le reste de ta méthode